sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -288,6 +289,11 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
           query,
           overwrite = true,
           ifPartitionNotExists)
+      case ctx: InsertIntoReplaceWhereContext =>
```

Comment:
I think we should fail if `identifierList` is specified. I'll update `OverwriteByExpression` to support column list later.

Reply:
Hmm, not fully understand but I think it might related to my previous comment :)

Reply:
Let me remove the identifierList from the scope of the current PR.